File: packages/coterie-py/src/coterie/core/llm/anthropic_client.py
"""Anthropic provider."""

from coterie.core.llm.base import LLMClient


class AnthropicClient(LLMClient):
    DEFAULT_MODEL = "claude-haiku-4-5-20251001"

    def __init__(self, model: str | None = None, *, max_tokens: int = 1200) -> None:
        try:
            import anthropic
        except ImportError as e:
            raise ImportError(
                "AnthropicClient requires `anthropic`. `pip install coterie[anthropic]` "
                "or the default install which includes it."
            ) from e
        self._client = anthropic.Anthropic()
        self.model = model or self.DEFAULT_MODEL
        self.max_tokens = max_tokens

    def chat(self, system: str, messages: list[dict]) -> str:
        resp = self._client.messages.create(
            model=self.model,
            max_tokens=self.max_tokens,
            system=system,
            messages=messages,
        )
        first = resp.content[0]
        return getattr(first, "text", str(first))

File: packages/coterie-js/src/core/llm/anthropicClient.ts
import type { LLMClient, LLMMessage } from "./base.js";

export class AnthropicClient implements LLMClient {
  static readonly DEFAULT_MODEL = "claude-haiku-4-5-20251001";

  private client: any;

  constructor(
    public readonly model: string = AnthropicClient.DEFAULT_MODEL,
    public readonly maxTokens: number = 1200,
  ) {}

  private async ensureClient(): Promise<any> {
    if (this.client) return this.client;
    let Anthropic: any;
    try {
      ({ default: Anthropic } = await import("@anthropic-ai/sdk"));
    } catch (e) {
      throw new Error(
        "AnthropicClient requires `@anthropic-ai/sdk`. Install with `npm install @anthropic-ai/sdk`.",
      );
    }
    this.client = new Anthropic();
    return this.client;
  }

  async chat(system: string, messages: LLMMessage[]): Promise<string> {
    const client = await this.ensureClient();
    const resp = await client.messages.create({
      model: this.model,
      max_tokens: this.maxTokens,
      system,
      messages,
    });
    const first = resp.content?.[0];
    return first?.text ?? String(first ?? "");
  }
}
